src/server/pfs/server/driver_fsck.go

```diff
@@ -377,18 +376,23 @@ func (d *driver) fsck(ctx context.Context, fix bool, cb func(*pfs.FsckResponse)
 	}
 	for _, repo := range repoInfos {
 		if err := dbutil.WithTx(ctx, d.env.DB, func(ctx context.Context, tx *pachsql.Tx) error {
-			commitInfo := &pfs.CommitInfo{}
-			if err := d.commits.ReadWrite(tx).GetByIndex(pfsdb.CommitsRepoIndex, pfsdb.RepoKey(repo.Repo), commitInfo, col.DefaultOptions(), func(string) error {
-				commitInfos[pfsdb.CommitKey(commitInfo.Commit)] = proto.Clone(commitInfo).(*pfs.CommitInfo)
-				return nil
-			}); err != nil {
+			commits, err := pfsdb.ListCommitTxByFilter(ctx, tx, pfsdb.CommitListFilter{pfsdb.CommitRepos: []string{pfsdb.RepoKey(repo.Repo)}}, false, false)
+			if err != nil {
 				return errors.Wrap(err, "get commits by commits repo index")
 			}
-			branchInfo := &pfs.BranchInfo{}
-			err := d.branches.ReadWrite(tx).GetByIndex(pfsdb.BranchesRepoIndex, pfsdb.RepoKey(repo.Repo), branchInfo, col.DefaultOptions(), func(string) error {
-				branchInfos[pfsdb.BranchKey(branchInfo.Branch)] = proto.Clone(branchInfo).(*pfs.BranchInfo)
+			for _, commitInfo := range commits {
+				commitInfos[pfsdb.CommitKey(commitInfo.Commit)] = commitInfo
+			}
+			iter, err := pfsdb.NewBranchIterator(ctx, tx, 0, 100, &pfs.Branch{Repo: repo.Repo})
```

Comment:
I don't think it's a good idea to expose an iterator that takes a transaction.  We avoided doing that for repos and commits.  Is this at least paging behind the scenes or does it hold open a `sql.Rows`?

Reply:
Something like `ListBranchTx(tx *sqlx.Tx, repoName string) ([]*BranchInfo, error)` would probably be harder to misuse.

Reply:
For repos, we do actually expose a way to create an iterator that takes a transaction and then iterate through the results:
```
// ListRepo returns a RepoIterator that exposes a Next() function for retrieving *pfs.RepoInfo references.
func ListRepo(ctx context.Context, tx *pachsql.Tx, filter RepoListFilter) (*RepoIterator, error)
```

Maybe that is something we should reconsider as part of the *cleanup* too? Do we want to have a ListRepoTx that does a streams.collect as it gets the repos? We could do that for branches and commits too

Reply:
Yeah that can be a cleanup item, doesn't need to block this PR.